terraform/windows.tf

```diff
@@ -1,3 +1,34 @@
 data "aws_ssm_parameter" "windows" {
   name = "/aws/service/ami-windows-latest/Windows_Server-2019-English-Core-EKS_Optimized-1.23/image_id"
 }
+
+locals {
+  windows_user_data = <<EOF
+<powershell>
+  # Set execution policy and update security protocol
+  Set-ExecutionPolicy Bypass -Scope Process -Force
+  [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.ServicePointManager]::SecurityProtocol -bor 3072
+
+  # Install Chocolatey
+  iex ((New-Object System.Net.WebClient).DownloadString('https://community.chocolatey.org/install.ps1'))
+
+  # Create directory for Bazel and download Bazelisk
+  $bazelDir = 'C:\bazel'
+  mkdir $bazelDir -Force
+  Invoke-WebRequest -Uri 'https://github.com/bazelbuild/bazelisk/releases/latest/download/bazelisk-windows-amd64.exe' -OutFile "$bazelDir\bazel.exe"
+
+  # Install aws cli
+  $command = "[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12"
+  Invoke-Expression $command
+  Invoke-WebRequest -Uri "https://awscli.amazonaws.com/AWSCLIV2.msi" -Outfile C:\AWSCLIV2.msi
+  $arguments = "/i `"C:\AWSCLIV2.msi`" /quiet"
+  Start-Process msiexec.exe -ArgumentList $arguments -Wait
+  $env:Path = [System.Environment]::GetEnvironmentVariable("Path","Machine")
+
+  # Install Git using Chocolatey
+  choco install -y git.portable
+  C:\tools\git\bin\git.exe clone https://github.com/envoyproxy/envoy.git C:/envoy
+</powershell>
+EOF
```

Comment:
re whole thing: where is this thing from? is it from Envoy build instructions?

Reply:
These steps are not from the envoy instruction; rather, they are the necessary actions to make building envoy possible. To avoid building it on Windows, I am using `./ci/run_envoy_docker.sh` for a more predictable build. Some tasks cannot be executed from PowerShell, or I am unsure how to do so. Therefore, I install the Git client to run bash and clone the repository. Additionally, we require the AWS client to update the binary because there is no SSH server, and the Windows AMI doesn't have enabled SSH features.